### PR TITLE
Update SPI README

### PR DESCRIPTION
**Pull request by axelclark**
* Spell out acronyms
* Add more explanation for reading data sheet

I've updated the expanded explanation to use the MCP3002.  I have one TODO I need some help with.  I couldn't figure out how to explain the MSBF configuration.

Let me know if you see any errors or would like me to make any changes.

### CODE REVIEW

File: README.md
Comment (axelclark) [reply README.md]: > Before it was 0x74 and 0x64. Then I changed it to 0x70 and 0x60. MSBF is the 0x08 bit, so it should have been low both times, right?
> The 0x04 bit should be in the don't care section, so that's why I got rid of it. Let me read the MSBF section. Maybe we do want it...
> Yeah, I think that we want it on, but I see why it doesn't matter. When it's off, it means that the MCP3002 outputs the value MSB first (the way we want), and then if we keep asking for data, we'll get the LSB version. If MSBF is on, then we couldn't keep clocking out data.
> We don't ask for more data, so we don't get the LSB version. Since CS goes high at the end of our transaction and then low at the beginning of the next, the chip is ok.
> This is a good catch. We should make it 0x78 and 0x68 to be completely correct.

**Comment by axelclark** [comment]
I updated the README based on your explanation on the MSBF from Slack.

**Comment by adkron** [comment]
I think even the links are a great improvement for helping people understand what this library is for and how to use it. This is "new to hardware" friendly. :+1:

**Review by adkron — APPROVED** [review]

File: README.md
Comment (fhunleth) [inline README.md]: I think that the sentence "An ADC is used to convert...so the signal can be read by the main processor" isn't needed since the previous sentences say this.

File: README.md
Comment (fhunleth) [inline README.md]: "Many processors, like the one on the Raspberry Pi, can't read analog signals directly,"

File: README.md
Comment (fhunleth) [inline README.md]: "The protocol" -> "It"

File: README.md
Comment (fhunleth) [inline README.md:156]: Good catch on the MSBF bit!

File: README.md
Comment (fhunleth) [inline README.md]: I'm for deleting this paragraph for now although I understand that this information is useful. In general, people should be adding their ElixirALE.SPI processes to supervision trees in their application. The `start_link` call above is shorthand for making an easy demo. Can we revisit this in a separate PR, since I think that the majority of this PR is really good and I don't want this to hold it up?

File: README.md
Comment (fhunleth) [inline README.md:183]: I almost thought that you were going to say "For more information on bitstrings, run `h <<>>` in IEx." or something similar. That might still be helpful somewhere else.

File: README.md
Comment (fhunleth) [inline README.md]: I never can get this URL to work for me. Does it work for you? My new advice is to run `h <<>>` at the IEx prompt.

**Review by fhunleth — CHANGES_REQUESTED** [review]
These changes are great! I think that most of it is ready to go with some really minor changes. The part that I'd like to hold off on is the additional detail part in the SPI section. That one we should chat about for how to integrate it in, but for now, I'd really like to get the other changes in to `master`. Thanks!

File: README.md
Comment (axelclark) [reply README.md]: Deleted the sentence.

File: README.md
Comment (axelclark) [reply README.md]: Updated

File: README.md
Comment (axelclark) [reply README.md]: Updated

File: README.md
Comment (axelclark) [reply README.md:156]: Thanks!

File: README.md
Comment (axelclark) [reply README.md:183]: Will use it below.

File: README.md
Comment (axelclark) [reply README.md]: Question:  Do you want to delete just this paragraph or the entire section discussing the `start_link` call?  I'm going to just delete this paragraph in my next update, but let me know if you mean the entire section.

File: README.md
Comment (axelclark) [reply README.md]: I tested it and it seems to work for me.  I'll add an "or run `h <<>>` at the IEx prompt."

**Comment by axelclark** [comment]
I made the requested changes.  The only question I had was on the `start_link` section.  I wasn't sure if you meant just that paragraph or the entire section.  I just deleted the paragraph, but can remove the entire section if that's what you meant.

Also, when I make updates, would you rather me leave them as a separate commit or use rebase to squash them into the single commit?  I used rebase to squash, but I also could have done that after you reviewed/approved the changes.

**Comment by fhunleth** [comment]
I just merged almost all of it in. I fixed some trailing whitespace and that had the effect of making the merge conflicts that you'll see now. I didn't pull in the paragraphs that describe the SPI example again, but in more detail. Not that I don't think that they're helpful, though. My thought is to integrate them differently. I copied the binary pattern matching note to the original example (that's committed). I think that the paragraph on finding the "spidev" device is useful as a FAQ (I didn't make that change.) The rest of it would be good for a blog post since it is super detailed on hardware and the Elixir. I'd add a link to it from here. I'm just not sure that I want to make the elixir_ale docs so detailed yet, and it looks out of balance since the GPIO and I2C sections aren't nearly as detailed.

I'm going to close this PR given how much was integrated. Thank you so much for making these doc updates!!

**Pull request closed without merging**
